File: 0100_Same_Tree/solution.java
package cn.huang.leetcode;
/*
    100  Same Tree
 */

import cn.huang.leetcode.common.TreeNode;

public class LC_0100_SameTree {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p == null && q ==null)
            return true;

        if(p == null || q == null)
            return false;

        return p.val == q.val && isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
    }
}

File: 0100_Same_Tree/solution.go
// https://leetcode.com/problems/same-tree/
package leetcode0100

import (
	. "../common"
)

func isSameTree(p *TreeNode, q *TreeNode) bool {
	if(p == nil && q == nil){
		return true
	}

	if(p==nil || q ==nil){
		return false
	}

	return p.Val == q.Val && isSameTree(p.Left, q.Left) && isSameTree(p.Right, q.Right)
}
